silence/silence.go

```diff
@@ -469,6 +471,20 @@ func (s *Silences) GC() (int, error) {
 
 // ValidateMatcher runs validation on the matcher name, type, and pattern.
 var ValidateMatcher = func(m *pb.Matcher) error {
+	return validateClassicMatcher(m)
```

Comment:
Please see my comment on types/types.go - I think we shoudl get rid of this global variable and use depency injection of the feature flagger to control how we validate these matchers.